src/storage/src/monitor/monitored_store.rs

```diff
@@ -222,7 +250,9 @@ impl<S: StateStore> StateStore for MonitoredStateStore<S> {
     }
 
     fn new_local(&self, table_id: TableId) -> Self::NewLocalFuture<'_> {
-        async move { MonitoredStateStore::new(self.inner.new_local(table_id).await, self.stats.clone()) }
+        async move {
```

Comment:
I think we can do `new` directly as what the current main does and remove the `MonitoredStateStore::new_local`. We can wrap the real inner (e.g. local hummock storage) with `TracedStateStore::new` in the `new_local` of `TracedStateStore`.